### PR TITLE
Creating a github-pages themed site for shadow-utils

### PR DESCRIPTION
**Pull request by AZaugg**
Initial commit of github-pages site, taking the contents of the
original site located at alioth.debian.org and moving it into
github pages.

https://github.com/shadow-maint/shadow/issues/114

### CODE REVIEW

**Comment by AZaugg** [comment]
When I was originally looking for this project I found lots of references to shadow-utils that were supposed to live at pkg-shadow.alioth.debian.org. It looks like this was shut down. I didn't think it would hurt to create a github-pages site to represent this project.  

https://docs.github.com/en/pages/quickstart

**Review by ikerexxe — COMMENTED** [review]
Good work! I've added several minor things to improve.

How can I test it? I've tried with `bundle exec jekyll server` but I get `Could not locate Gemfile or .bundle/ directory`. Maybe you forgot to add the Gemfile?

**Comment by jubalh** [comment]
Wouldn't it be nicer to have this in a separate repo under the `shadow-maint` organization?

File: docs/_site/feed.xml [hunk 1]
```diff
@@ -0,0 +1 @@
+<?xml version="1.0" encoding="utf-8"?><feed xmlns="http://www.w3.org/2005/Atom" ><generator uri="https://jekyllrb.com/" version="4.2.1">Jekyll</generator><link href="http://0.0.0.0:4000/feed.xml" rel="self" type="application/atom+xml" /><link href="http://0.0.0.0:4000/" rel="alternate" type="text/html" /><updated>2022-01-11T23:07:48-05:00</updated><id>http://0.0.0.0:4000/feed.xml</id><title type="html">shadow-utils</title><subtitle>This is the shadow-utils tool suite home page.</subtitle></feed>
```
[the remark this answers is not in the document]
Comment (AZaugg) [reply docs/_site/feed.xml]: Added new line

File: docs/getinvolved.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+</ul>
+
+<p>Bug reports, patches, feature requests or any feedback are welcomed and can be created <a href="https://github.com/shadow-maint/shadow/issues">here</a>.</p>
+
+
+<hr/>
+<h2>Testsuite</h2>
+You can also contribute to the test suite and improve the coverage of these tests.
+
+The testsuite is currently executed on a Debain environment. You can help testing it on another environment.
```
[the remark this answers is not in the document]
Comment (AZaugg) [reply docs/getinvolved.md]: Fixed

**Comment by AZaugg** [comment]
Thanks for taking a look @ikerexxe , I added the Gem file. To test locally, I locally installed jekyll locally
$ gem install --user-install bundler jekyll
$ bundle exec jekyll serve  --watch 

When testing locally all the metadata that is pulled in via site.github won't render.

@jubalh i placed it into the same repos so i could make reference to the metadata stored in site.github, I pull the following metadata from github:
- latest version
- Contributors
- all releases

Let me know if you don't see value and I can move it else where. I think we will also need the person that owns the repo to enable githubpages for this repo.

File: docs/Gemfile [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+#     bundle exec jekyll serve
+#
+# This will help ensure the proper Jekyll version is running.
+# Happy Jekylling!
+gem "jekyll", "~> 4.2.1"
+# This is the default theme for new Jekyll sites. You may change this to anything you like.
+gem "minima", "~> 2.5"
+# If you want to use GitHub Pages, remove the "gem "jekyll"" above and
+# uncomment the line below. To upgrade, run `bundle update github-pages`.
+# gem "github-pages", group: :jekyll_plugins
```
Comment (ikerexxe) [inline docs/Gemfile:15]: As per comments in the file to generate a github-page you need to: `gem "jekyll", "~> 4.2.1"` should be commented and `gem "github-pages", group: :jekyll_plugins` uncommented.

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+<hr/>
+
+<h2>Latest release</h2>
+Latest Version release: {{ site.github.releases_url }}
+
+<hr/>
+
+<h2>Issues</h2>
+Open issues can be found <a href="https://github.com/shadow-maint/shadow/issues?page=2&q=is%3Aissue+is%3Aopen">here</a>
```
Comment (ikerexxe) [inline docs/index.md:19]: Add an ending period after `here`

File: docs/downloads.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+---
+layout: page
+title: "Downloads"
+permalink: /downloads/
+---
+
+<h2>Git repos</h2>
+On going development is done on <a href="https://github.com/shadow-maint/shadow">github</a>.
```
Comment (ikerexxe) [inline docs/downloads.md:8]: I think that it's spelled "ongoing" but I'm not a native English speaker so I'm not completely sure.

Note: https://english.stackexchange.com/questions/9240/ongoing-or-on-going seems to agree.

File: docs/getinvolved.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+You can also contribute to the test suite and improve the coverage of these tests.
+
+The testsuite is currently executed on a Debian environment. You can help testing it on another environment.
+
+
+<h2>Authors and contributors</h2>
+
+{% for contributors in site.github.contributors %}
+  * [{{ contributors.login }}]({{contributors.contributors_url}})
+{% endfor %}
```
Comment (ikerexxe) [inline docs/getinvolved.md:35]: These lines generate a list with the contributors with a link, but the links point to the home page of the project. Is this intentional?

**Review by ikerexxe — COMMENTED** [review]
Now that I was able to generate the web page I have been able to review it better.

I wonder, do we need to track the md and html files in git? Isn't it enough to track the md files?

**Comment by hallyn** [comment]
Hi,

note that 

http://shadow-maint.github.io/shadow-www/

is the github-pages site for

https://github.com/shadow-maint/shadow-www

**Comment by hallyn** [comment]
> Wouldn't it be nicer to have this in a separate repo under the `shadow-maint` organization?

Yes, this should be done as PRs against https://github.com/shadow-maint/shadow-www, right?  Or does this serve a different purpose?

**Pull request closed without merging**